File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Mapper.Model/MassimariScarto/Implementation/MassimarioScartoModelMapper.cs
using VecompSoftware.DocSuiteWeb.Entity.MassimariScarto;
using VecompSoftware.DocSuiteWeb.Model.Entities.MassimariScarto;

namespace VecompSoftware.DocSuiteWeb.Mapper.Model.MassimariScarto
{
    public class MassimarioScartoModelMapper : BaseModelMapper<MassimarioScarto, MassimarioScartoModel>, IMassimarioScartoModelMapper
    {
        #region [ Fields ]

        #endregion

        #region [ Constructor ]
        public MassimarioScartoModelMapper()
        {

        }
        #endregion

        #region [ Methods ]
        public static DocSuiteWeb.Model.Entities.MassimariScarto.MassimarioScartoStatus StatusConverter(Entity.MassimariScarto.MassimarioScartoStatus status)
        {
            switch (status)
            {
                case Entity.MassimariScarto.MassimarioScartoStatus.Active:
                    return DocSuiteWeb.Model.Entities.MassimariScarto.MassimarioScartoStatus.Active;
                case Entity.MassimariScarto.MassimarioScartoStatus.LogicalDelete:
                    return DocSuiteWeb.Model.Entities.MassimariScarto.MassimarioScartoStatus.LogicalDelete;
                default:
                    return DocSuiteWeb.Model.Entities.MassimariScarto.MassimarioScartoStatus.Active;
            }
        }

        public override MassimarioScartoModel Map(MassimarioScarto entity, MassimarioScartoModel modelTransformed)
        {
            modelTransformed.UniqueId = entity.UniqueId;
            modelTransformed.Code = entity.Code;
            modelTransformed.FullCode = entity.FullCode;
            modelTransformed.ConservationPeriod = entity.ConservationPeriod;
            modelTransformed.EndDate = entity.EndDate;
            modelTransformed.LastChangedDate = entity.LastChangedDate;
            modelTransformed.LastChangedUser = entity.LastChangedUser;
            modelTransformed.MassimarioScartoLevel = entity.MassimarioScartoLevel;
            modelTransformed.MassimarioScartoPath = entity.MassimarioScartoPath;
            modelTransformed.MassimarioScartoParentPath = entity.MassimarioScartoParentPath;
            modelTransformed.Name = entity.Name;
            modelTransformed.Note = entity.Note;
            modelTransformed.FakeInsertId = entity.FakeInsertId;
            modelTransformed.RegistrationDate = entity.RegistrationDate;
            modelTransformed.RegistrationUser = entity.RegistrationUser;
            modelTransformed.StartDate = entity.StartDate;
            modelTransformed.Status = StatusConverter(entity.Status);
            return modelTransformed;
        }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Mappers/MassimariScarto/MassimarioScartoModelMapper.ts
import BaseMapper = require('App/Mappers/BaseMapper');
import MassimarioScartoModel = require('App/Models/MassimariScarto/MassimarioScartoModel');
import MassimarioScartoStatusType = require('App/Models/MassimariScarto/MassimarioScartoStatusType');

class MassimarioScartoModelMapper extends BaseMapper<MassimarioScartoModel> {

    constructor() {
        super();
    }

    public Map(source: any): MassimarioScartoModel {
        let toMap: MassimarioScartoModel = new MassimarioScartoModel();

        if (!source) {
            return null;
        }

        toMap.UniqueId = source.UniqueId;
        toMap.Status = source.Status;
        toMap.Name = source.Name;
        toMap.Code = source.Code;
        toMap.FullCode = source.FullCode;
        toMap.Note = source.Note;
        toMap.ConservationPeriod = source.ConservationPeriod;
        toMap.StartDate = source.StartDate;
        toMap.EndDate = source.EndDate;
        toMap.MassimarioScartoPath = source.MassimarioScartoPath;
        toMap.MassimarioScartoParentPath = source.MassimarioScartoParentPath;
        toMap.MassimarioScartoLevel = source.MassimarioScartoLevel;
        toMap.FakeInsertId = source.FakeInsertId;
        toMap.RegistrationUser = source.RegistrationUser;
        toMap.RegistrationDate = source.RegistrationDate;
        toMap.LastChangedUser = source.LastChangedUser;
        toMap.LastChangedDate = source.LastChangedDate;        

        return toMap;
    }

}

export = MassimarioScartoModelMapper;